blog/2023-11-06-idp-data-preso/index.md

```diff
@@ -0,0 +1,87 @@
+
+Next we discuss these approaches in more details.
+
+## Use your existing CI/CD pipelines to construct the source of truth
+Starting from the specification of entities and infrastructure resources in a
+git repository, this approach utilizes tasks in the CI to hydrate entities with
+information on entity relations, extra metadata, and status of deployed
+resources. To avoid conflating user changes and automated CI changes, the
+hydrated entities are often kept in a separate git repository that mirrors and
+expands entities in the original git repository with intended application
+specifications.
+
+On the positive side:
+
+* This is a relatively simple approach and works for smaller teams with smaller number of applications or systems
+* Having a second git repository to capture the end state of an entity stays closer to the core GitOps practices
+* Does not require significant modification to the developer portal
+
+On the negative side:
+
+* There is inherent duplications that are happening
+* Adding custom metadata by application teams is not as trivial as it requires making changes to the integration workflow, thus bringing load and demand to the DevOps teams
+* Less abstraction in place as end application users are directly exposed to the yaml specification of the entities
+* Does not scale well as the number of systems and entities grow
+
+## Use a central control plane as the source of truth
+
+The hub and spoke model is the most advocated for model when applying GitOps practices. Your control plane cluster runs and manages your platform tools, your CI, your CD, developer portal, infrastructure as code tooling, etc.
+
+On the positive side:
+
+* There really is a single place to inspect the status of entities. E.g., Argo applications can tell you the status of deployed applications. You can also inspect the status of workflows, infrastructure resources, and any other entity that the control plane cluster manages.
+* You can use the Backstage Kubernetes plugin seamlessly and maybe with some little tweaks or with introducing fairly light-weight Backstage custom entity provides pull and show the status of entities in the Backstage portal.
```

Comment:
The currently reads a bit confusing -- not sure if there are missing or incorrect words?

Reply:
thanks, rephrasing.